mock/py/mockbuild/util.py

```diff
@@ -1356,6 +1360,16 @@ def include(config_file, config_opts, is_statement=False):
         raise exception.ConfigError("Could not find included config file: %s" % config_file)
 
 
+@traceLog()
+def find_config_in_path(config_file, config_dir):
+    path = [config_dir, os.path.join(config_dir, "templates"), "/etc/mock/templates/"]
```

Comment:
I'm not able to tell this is the best approach TBH, the fact that we include relatively to the first chosen config file (the main chroot config) brings non-determinism to mock's behavior (the includes happen differently, depending where the config resides).

Usually there's some default search path (notably $PATH but also sys.path for python modules), and you know where you include from - no matter where the input file is, or what is the $PWD.  Well the `sys.path` has flaw that it contains the `pwd`, compared to $PATH (and other *PATH vars), but still.

Reply:
I get the point and kind of agree. There are basically three use-cases though.

1. Mock configs are in the `/etc/mock` and I just want to use them
2. I am hacking on the configs and want to use those from git directory
3. I have my own personal configs somewhere in my system and I don't necessarily have to use `templates` directory for storing templates.

The current implementation deals with the first two cases with ease and partially solves the third one. If we would go with the proposed solution and set the `path` to some locations, not containing those relative to the specified config, we won't IMHO deal even with the case 2 - configs from git (Only if we would have `pwd` in the path and move ourselves directly to the `~/whatever/mock/mock-core-configs/etc/mock/templates`). We would have to read some environment variable like `MOCKCONFPATH` to allow us explicitly specify the possible config locations. Are you suggesting this, or am I unnecessarily complicating the issue?

Reply:
> Are you suggesting this

Yes.  Either commandline option, or env var (but env var is sort of standard for *PATH use-case).

Reply:
I would vote for `include('templates/centos-stream.tpl')` and search only in config dirs (we have several of them). This way we would not need to have `templates` directory hard coded in code and it will solve the third case really nice.

Reply:
I agree with @xsuchy, the `templates` doesn't have to be there.  But for 2 we still need to have a way to adjust the search patch at runtime.

> we have several of them

Hmm, I thought there's only /etc/mock.

Reply:
`~/.config/mock/` as well. 

```
       -r CONFIG, --root=CONFIG
              Uses  specified  chroot  configuration  as  defined  in  ~/.config/mock/<CONFIG>.cfg or /etc/mock/<CON‐
              FIG>.cfg.  Optionally if CONFIG ends in '.cfg', it is interpreted as full path to config file. If  none
              specified, uses the chroot config linked to by /etc/mock/default.cfg.
```

Reply:
And technically it can be in https://github.com/rpm-software-management/mock/wiki#generate-custom-config-file